### PR TITLE
Update unit tech advancement when preview refreshes.

### LINKED ISSUE #466: Preview summary screen does not update era availability
### Environment
MML 0.47.4 (stand alone)
Java version 1.8.0_241
Platform Windows 10 10.0 (x86)

### Description
While updating a mech design I noticed the Preview screen does not update The level / era data. In my original design I had protype jump jets which have an extinct era as 3069. I changed to normal jump jests and when I viewed the Preview page the extinct era was still 3069. 

When I loaded the mech in MHQ the extinct era was correct.

### PR DESCRIPTION
**Pull request by neoancient**
Recalculate the composite tech advancement and availability codes for the unit based on parts and construction options. Add a preview refresh when tech base/tech level/year changes for those unit types that don't already do it.

Fixes #466

### CODE REVIEW

File: src/megameklab/com/ui/tabs/PreviewTab.java [hunk 1]
```diff
@@ -59,6 +59,7 @@ public PreviewTab(EntitySource eSource) {
 	public void refresh() {
 		boolean populateTextFields = true;
 		final Entity selectedUnit = eSource.getEntity();
+		selectedUnit.recalculateTechAdvancement();
         MechView mechView = null;
```
Comment (Windchild292) [inline src/megameklab/com/ui/tabs/PreviewTab.java]: Does this formatting look weird to anyone else?

File: src/megameklab/com/util/RefreshListener.java [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+package megameklab.com.util;
+
+import java.util.EventListener;
+
+public interface RefreshListener extends EventListener{
```
Comment (Windchild292) [inline src/megameklab/com/util/RefreshListener.java]: Why does this entire file show as changed? Spacing/file save issues?

Comment (neoancient) [reply src/megameklab/com/util/RefreshListener.java]: Usually that means the line endings were CR/LF and they got changed to LF. GitHub lets you hide whitespace changes when you view the diff so you can see what actually changed.

File: src/megameklab/com/ui/tabs/PreviewTab.java (hunk 1 above)
Comment (neoancient) [reply src/megameklab/com/ui/tabs/PreviewTab.java]: Those lines were tab-indented. Fixed now.

**Pull request merged**